osu.Server.Spectator/Hubs/ServerMultiplayerRoom.cs

```diff
@@ -27,13 +29,22 @@ public MatchTypeImplementation MatchTypeImplementation
             }
         }
 
-        public ServerMultiplayerRoom(long roomId, IMultiplayerServerMatchCallbacks hubCallbacks)
+        public readonly MultiplayerQueue QueueImplementation;
+
+        public ServerMultiplayerRoom(long roomId, IDatabaseFactory dbFactory, IMultiplayerServerMatchCallbacks hubCallbacks)
             : base(roomId)
         {
             this.hubCallbacks = hubCallbacks;
 
             // just to ensure non-null.
             matchTypeImplementation = createTypeImplementation(MatchType.HeadToHead);
+            QueueImplementation = new MultiplayerQueue(this, dbFactory, hubCallbacks);
+        }
+
+        public async Task Initialise()
```

Comment:
Is this split out from the constructor for the sole purpose of setting properties to their correct values before calling?

Wondering if it could potentially be moved to the `Settings` setter?

Reply:
See answer below.